tests/api-tests/queries/singletons.test.ts

```diff
@@ -0,0 +1,210 @@
+
+      const { errors, data } = await context.graphql.raw({
+        query: `mutation { updateSingular(data: { field: "something here" }) { id, field } }`,
+      });
+
+      expect(errors).toEqual(undefined);
+      expect(data).toEqual({ updateSingular: { field: 'something here', id: '1' } });
+    })
+  );
+  test(
+    'delete',
+    runner(async ({ context }) => {
+      await initialiseData({ context });
+
+      const { data, errors } = await context.graphql.raw({
+        query: `mutation { deleteSingular { id, field } }`,
+      });
+
+      expect(errors).toEqual(undefined);
+      expect(data).toEqual({ deleteSingular: { field: '', id: '1' } });
+
+      const { data: queryData, errors: readErrors } = await context.graphql.raw({
+        query: `query { singular { id } }`,
+      });
+
+      expect(readErrors).toEqual(undefined);
+      expect(queryData).toEqual({ singular: null });
+    })
+  );
+  test(
+    'fails to create when existing item',
+    runner(async ({ context }) => {
+      await initialiseData({ context });
+
+      const { errors } = await context.graphql.raw({
+        query: `mutation { createSingular(data: { field: "Something here" }) { id } }`,
+      });
+
+      expect(errors?.[0].message).toEqual(
+        'Prisma error: Unique constraint failed on the fields: (`id`)'
+      );
+    })
+  );
+  test(
+    'fails to create with differing ID',
+    runner(async ({ context }) => {
+      const { errors } = await context.graphql.raw({
+        query: `mutation { createSingular(data: { id: 2, field: "Something here" }) { id } }`,
+      });
+
+      expect(errors?.[0].message).toEqual(
+        `Field \"id\" is not defined by type \"SingularCreateInput\".`
+      );
+    })
+  );
+  // Testing these for certainty - nobody should be using these
+  test(
+    'create many',
+    runner(async ({ context }) => {
+      const { data, errors } = await context.graphql.raw({
+        query: `mutation { createSingulars(data: [{ field: "Something here" }]) { id, field } }`,
+      });
+
+      expect(errors).toEqual(undefined);
+      expect(data).toEqual({ createSingulars: [{ field: 'Something here', id: '1' }] });
+    })
+  );
+  test(
+    'update many',
+    runner(async ({ context }) => {
+      await initialiseData({ context });
+
+      const { data, errors } = await context.graphql.raw({
+        query: `mutation { updateSingulars(data: [{ data: { field: "Something here" } }]) { id, field } }`,
+      });
+
+      expect(errors).toEqual(undefined);
+      expect(data).toEqual({ updateSingulars: [{ field: 'Something here', id: '1' }] });
+    })
+  );
+  test(
+    'delete many',
+    runner(async ({ context }) => {
+      await initialiseData({ context });
+
+      const { data, errors } = await context.graphql.raw({
+        query: `mutation { deleteSingulars(where: { id: 1 }) { id, field } }`,
+      });
+
+      expect(errors).toEqual(undefined);
```

Comment:
```suggestion
      const data = await context.graphql.run({
        query: `mutation { deleteSingulars(where: { id: 1 }) { id, field } }`,
      });

```

I don't really care about this too much and don't mind if you want to change it or not and I know we have a bunch of tests that don't work like this but ideally, we should use `context.graphql.run` here.